module1/homework/make_unique/make_unique.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <type_traits>
+
+namespace cs {
+
+template <typename Type, class... Args>
+std::enable_if_t<!std::is_array_v<Type>, std::unique_ptr<Type>>
+make_unique(Args&&... args) {
+    return std::unique_ptr<Type>(new Type(std::forward<Args>(args)...));
+}
+
+template<typename Type>
+std::unique_ptr<Type> make_unique(std::size_t size) {
+    return std::unique_ptr<Type>(new typename std::remove_extent<Type>::type[size]());
```

Comment:
Doesn't your implementation leak because of that?

```suggestion
    return std::unique_ptr<Type[]>(new typename std::remove_extent<Type>::type[size]());
```

Reply:
nope, valgrind says: "all heaps blocks were freed"
what's more with suggested change i get error creating array of int []

Reply:
That's right, my bad. Compiler automaticaly deduces T as an array, eg. int[], so there is no need to add []